File: LivrusVirtus/FoxToolKit/FXPython/ConProfissoes.py
from FXPy.fox import *
from Consultas import frmConsultas
from resource import Resource
from ConsultasSQL import ConsultasSQL
import Principal

class frmConProfissoes(frmConsultas):

  def __init__(self, owner):
    frmConsultas.__init__(self, owner, Resource.STR_CONPROFISS)  

    FXMAPFUNC(self, SEL_COMMAND, frmConProfissoes.ID_FRMCONSULTAS_BTNPESQUISAR, 
      frmConProfissoes.onBtnPesquisar)
    FXMAPFUNC(self, SEL_COMMAND, frmConProfissoes.ID_FRMCONSULTAS_BTNLIMPAR, 
      frmConProfissoes.onBtnLimpar)
    FXMAPFUNC(self, SEL_SELECTED, frmConProfissoes.ID_FRMCONSULTAS_GRIDCONSULTA, 
      frmConProfissoes.onGridSelected)
    FXMAPFUNC(self, SEL_CHANGED, frmConProfissoes.ID_FRMCONSULTAS_GRIDCONSULTA, 
      frmConProfissoes.onGridChanged)
    
    self.labDescricao.setText("Profissão:")
    self.gridConsulta.setItemText(0, 2, "Profissão")

  def create(self):
    frmConsultas.create(self)
  
  def onBtnPesquisar(self, sender, sel, ptr):
    self.PesquisarDados(ConsultasSQL.ConsSQL.Profissao('S',"", 
      self.edtDescricao.getText()), Resource.STR_PROFNENC, 2)
    return True
  
  def onBtnLimpar(self, sender, sel, ptr):
    self.LimparDados()
    return True
  
  def onGridSelected(self, sender, sel, ptr):
    self.setLabRegistros(sender.getCurrentRow())
    return True
  
  def onGridChanged(self, sender, sel, ptr):
    self.setLabRegistros(sender.getCurrentRow())
    return True

File: LivrusVirtus/VCPP6/FoxToolKit/LibPostgreSQL/ConProfissoes.cpp
#include "resource.h"
#include "ConProfissoes.h"
#include "Principal.h"
#include "ConsultasSQL.h"

CfrmConProfissoes *frmConProfissoes = (CfrmConProfissoes *) NULL;

FXDEFMAP(CfrmConProfissoes) CfrmConProfissoesMap[]={
  FXMAPFUNC(SEL_COMMAND, CfrmConProfissoes::ID_FRMCONSULTAS_BTNPESQUISAR, 
    CfrmConProfissoes::onBtnPesquisar),
  FXMAPFUNC(SEL_COMMAND, CfrmConProfissoes::ID_FRMCONSULTAS_BTNLIMPAR, 
    CfrmConProfissoes::onBtnLimpar),
  FXMAPFUNC(SEL_SELECTED,CfrmConProfissoes::ID_FRMCONSULTAS_GRIDCONSULTA,
    CfrmConProfissoes::onGridSelected),
  FXMAPFUNC(SEL_CHANGED,CfrmConProfissoes::ID_FRMCONSULTAS_GRIDCONSULTA,
    CfrmConProfissoes::onGridChanged)
};
FXIMPLEMENT(CfrmConProfissoes,CfrmConsultas,CfrmConProfissoesMap,
  ARRAYNUMBER(CfrmConProfissoesMap))

CfrmConProfissoes::CfrmConProfissoes(FXWindow* owner):
  CfrmConsultas(owner, STR_CONPROFISS) {
  IniciaControles();
}

void CfrmConProfissoes::IniciaControles() {
  this->labDescricao->setText("Profissão:");
  this->gridConsulta->setColumnText(1, "Profissão");
}

void CfrmConProfissoes::create(){
  FXDialogBox::create();
  show(PLACEMENT_SCREEN);
}

CfrmConProfissoes::~CfrmConProfissoes(){  
}

long CfrmConProfissoes::onBtnPesquisar(FXObject*,FXSelector,void*) {
  PesquisarDados(
    ConsSQL->Profissao('S',"", edtDescricao->getText()), STR_PROFNENC, 2);
  return true;
}

long CfrmConProfissoes::onBtnLimpar(FXObject*,FXSelector,void*) {
  LimparDados();
  return true;
}

long CfrmConProfissoes::onGridSelected(FXObject*,FXSelector,void* ptr){
  FXTableRange *escala = (FXTableRange*) ptr;
  setLabRegistros(escala->fm.row);
  return true;
}

long CfrmConProfissoes::onGridChanged(FXObject*,FXSelector,void* ptr){
  FXTablePos *pos = (FXTablePos*) ptr;
  setLabRegistros(pos->row);
  return true;
}
